app/presenters/tree_builder_services.rb

```diff
@@ -21,16 +21,12 @@ def root_options
 
   # Get root nodes count/array for explorer tree
   def x_get_tree_roots(count_only, _options)
-    roots = Rbac.filtered(Service.roots)
-
-    # Preload the root service picture since it's called by TreeNodeBuilder#build
-    MiqPreloader.preload(roots, :picture)
-
-    count_only_or_objects(count_only, roots, "name")
-  end
-
-  def x_get_tree_service_kids(object, count_only)
-    objects = Rbac.filtered(object.direct_service_children.select(&:display).sort_by { |o| o.name.downcase })
-    count_only_or_objects(count_only, objects, 'name')
+    all_services = Rbac.filtered(Service.where(:display => true))
+    if count_only
+      all_services.size
+    else
+      MiqPreloader.preload(all_services.to_a, :picture)
+      Service.arrange_nodes(all_services.sort_by { |n| [n.ancestry.to_s, n.name.downcase] })
```

Comment:
This is fine, so lets just leave it.

This might be a simple lack of a index on the `:name` column, though, I don't know if that is used in a sort or not.  Regardless, it isn't enough of a performance loss/gain compared to the entirety of this PR to warrant any further looking into this.  We can circle back if it becomes a problem, and it is an easy swap in and out.